MusicCat-Console/Program.cs

```diff
@@ -52,5 +64,40 @@ static void Main(string[] args)
 			logStream?.Dispose();
             Listener.Stop();
         }
+
+		private static int counter;
+		private static void OnConfigChange(object sender, FileSystemEventArgs e)
+		{
+			if (counter == 0)
+			{
+				counter++;
+				return;
+			}
+
+			counter = 0;
+			try
+			{
+				logWriter?.Dispose();
+				logStream?.Dispose();
+				Listener.Config = Config.ParseConfig(out logStream, out logWriter);
+			}
+			catch (Exception ex)
+			{
+				Log(new ApiLogMessage($"Failed to de-serialize config, using default config instead. Exception: {ex.Message}{Environment.NewLine}{ex.StackTrace}", ApiLogLevel.Warning));
+				Listener.Config = Config.DefaultConfig;
+			}
+			Listener.Stop();
+			Listener.Start();
+			try
+			{
+				MetadataStore.SongList.Clear();
+				MetadataStore.LoadMetadata(Log);
+			}
+			catch (Exception ex)
+			{
+				Log(new ApiLogMessage($"Failed to load metadata. Exception: {ex.Message}{Environment.NewLine}{ex.StackTrace}", ApiLogLevel.Critical));
+				Environment.Exit(1);
+			}
+		}
```

Comment:
This logic doesn't belong here. The console and service wrappers should be as empty as possible.

Reply:
where would you have the logic be instead?

Reply:
Inside MusicCat itself. Create a manager class inside MusicCat that the wrappers create, pass the config to, and start. Ideally, the service and console wrappers should handle finding the config and starting the service. Anything  that you copy and paste between both wrappers is likely better suited for MusicCat itself.